ddtrace/runtime_metrics/runtime_metrics.py

```diff
@@ -0,0 +1,138 @@
+import logging
+import threading
+import time
+
+from .metric_collectors import (
+    GCRuntimeMetricCollector,
+    PSUtilRuntimeMetricCollector,
+)
+
+AGENT_HOST = '127.0.0.1'
+DD_METRIC_PREFIX = 'runtime.python'
+FLUSH_INTERVAL = 10
+METRIC_AGENT_PORT = 8125
+
+
+log = logging.getLogger(__name__)
+
+
+# Default metrics to collect
+ENABLED_METRICS = set([
+    'ctx_switch.voluntary',
+    'ctx_switch.involuntary',
+    'cpu.time.sys',
+    'cpu.time.user',
+    'cpu.percent',
+    'gc.gen1_count',
+    'gc.gen2_count',
+    'gc.gen3_count',
+    'mem.rss',
+    'thread_count',
+])
+
+ENABLED_TAGS = set([
+    'service',
+    'runtime-id',
+])
+
+
+class RuntimeMetricsCollector(object):
+    """
+    TODO: configuration
+    """
+
+    METRIC_COLLECTORS = [
+        PSUtilRuntimeMetricCollector(),
+        GCRuntimeMetricCollector(),
+    ]
+
+    TAG_COLLECTORS = [
+    ]
+
+    def __init__(self, runtime_id, enabled_metrics=ENABLED_METRICS, enabled_tags=ENABLED_TAGS):
+        self._agent_host = AGENT_HOST
+        self._agent_metric_port = METRIC_AGENT_PORT
+        self.enabled_metrics = enabled_metrics
+        self.enabled_tags = enabled_tags
+        self._statsd = None
+        self._tracer_tags = [
+            self._metric_tag('runtime-id', runtime_id),
```

Comment:
Should this be a metric tag or a normal tag? Is there a difference in `dd-trace-py`?

Reply:
This is a tag for a metric, not a metric tag on a trace.

In either case they should be "normal tags"

Reply:
hahah now we have two different meanings for both tags and metrics... 😅